### PR TITLE
Introduce interoperability test program for Dust DDS 0.11

### PR DESCRIPTION
**Pull request by stkimmer**
Add the source code and build files for the interoperability executable to be run with other DDS implementations.  

The executable did run successfully against Connext DDS while all tests were passing.
The report and the steps that were executed can be seen here: https://github.com/s2e-systems/dust-dds/actions/runs/10385346755/job/28754286361#step:14:20 

should solve a step of #49

### CODE REVIEW

File: srcRs/DustDDS/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+            panic!("Ownership strength must be positive or zero")
+        }
+        OwnershipStrengthQosPolicy {
+            value: self.ownership_strength,
+        }
+    }
+}
+
+#[derive(Debug, dust_dds::topic_definition::type_support::DdsType)]
+pub struct ShapeType {
```
Comment (angelrti) [inline srcRs/DustDDS/src/main.rs:251]: Is there a way of providing the ShapeType as an IDL file?

**Review by angelrti — CHANGES_REQUESTED** [review]

File: srcRs/DustDDS/src/main.rs (hunk 1 above)
Comment (angelrti) [reply srcRs/DustDDS/src/main.rs:251]: My knowledge about rust is limited, so I only have one comment about the type definition

**Comment by angelrti** [comment]
There are a few other changes that we need to do, but we could do them in a different PR:

- Add copyright: https://github.com/omg-dds/dds-rtps/blob/master/COPYRIGHT.md
- Company name: https://github.com/omg-dds/dds-rtps/blob/master/generate_xlsx_report.py#L54
- Product name: https://github.com/omg-dds/dds-rtps/blob/master/generate_xlsx_report.py#L68
- Consideration per product: https://github.com/omg-dds/dds-rtps/blob/master/doc/test_description.template.rst?plain=1#L49
- Copyright for the report: https://github.com/omg-dds/dds-rtps/blob/master/doc/vars.rst#L17

File: srcRs/DustDDS/src/main.rs (hunk 1 above)
Comment (stkimmer) [reply srcRs/DustDDS/src/main.rs:251]: That is actually possible. We could use the idl that is already present in the srcCxx folder.

**Review by angelrti — APPROVED** [review]

**Pull request merged**